Fix signed assertions

explunit opened the pull request:
Fixes #191 and moves in the direction mentioned in #137

explunit commented on Kentor.AuthServices/IdentityProvider.cs:
```diff
@@ -262,6 +263,19 @@ public AsymmetricAlgorithm SigningKey
+                return certificate;
+            }
+        }
+
```
I'm not happy with this part, but could not figure out another way to get the info needed by the TokenResolver. If there's a way to construct `SecurityToken` from just the `SigningKey`, let me know.

explunit commented on Kentor.AuthServices.TestHelpers/SignedXmlHelper.cs:
```diff
@@ -21,7 +21,7 @@ public class SignedXmlHelper
         public static string SignXml(string xml)
         {
-            var xmlDoc = new XmlDocument { PreserveWhitespace = true };
+            var xmlDoc = new XmlDocument { PreserveWhitespace = false };
```
Even after fixing the main issue with TokenResolver, this change was needed to suppress a "Digest Validation Failed" error in `Saml2SecurityTokenHandler.ReadToken`. The whitespace from the test cases was confusing WIF somehow.

AndersAbel commented:
Thanks a lot for this (and your other PRs). I've been terribly busy the last few weeks and haven't had time to look at it. I'll review these as soon as possible, next week looks less busy.

explunit commented:
@AndersAbel I fixed the issue (and added test case) with Idp's that have only metadata Url and no local certificate. 

However, the problem remains with overwriting the IssuerTokenResolver on the shared configuration.
WIF ignores any issuerToken passed into the `readToken` method, and even within `readAssertion` it is using the value from Configuration, so creating our own version of `readToken` will not solve it:
http://referencesource.microsoft.com/#System.IdentityModel/System/IdentityModel/Tokens/Saml2SecurityTokenHandler.cs,2644

One solution might be to construct a new token handler during each request rather than using a shared instance, but I'm not sure of the performance impact.

explunit commented:
Closing this for now, as it will conflict with other changes that @AndersAbel has in progress
